src/core/parse_worker.cpp

```diff
@@ -0,0 +1,103 @@
+#include "parse_worker.hpp"
+
+namespace docgen {
+namespace core {
+
+/*
+ * Processes based on the current handler.
+ */
+void ParseWorker::proc(const token_t& t)
+{
+	if (done_()) {
+		return;
+	}
+
+	if (handler_().match(t.name)) {
+		handler_().handle_token(t);
+
+		handler_().reset_workers();
+
+		++handler_i_;
+
+		if (done_()) {
+			if (indefinite_() || itered_++ < iters_) {
```

Comment:
Shouldn't it be `++itered_ < iters_`? Otherwise woudn't it iterate 1 too many times

Reply:
Haha u right